{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "CONTRIBUTOR",
    "body": "### Problem\n\nThe display for a `vim.on_key()` callback error does not have a stack traceback.\n```\nError executing  vim.on_key Lua callback: vim/_editor.lua:0: Error executing 'on_key' with ns_ids '14'\n    Messages: [string \":source (no file)\"]:2: in F2()\nstack traceback:\n        [C]: in function 'error'\n        vim/_editor.lua: in function <vim/_editor.lua:0>\n```\n\nI don't know if this is a general problem with callbacks or specific to `vim.on_key()`\n\n### Steps to reproduce\n\nsource the following and  hit any key.\n```lua\nlocal function F2()\n    error(\"in F2()\")\nend\nlocal function F1()\n    F2()\nend\nlocal function handler()\n    F1()\nend\nvim.on_key(handler)\n```\n\n### Expected behavior\n\nCalling `handler` directly produces the following. That's a good place to start.\n```\nstack traceback:                                                                                    \n        [C]: in function 'error'                                                                    \n        [string \":source (no file)\"]:6: in function 'F2'                                            \n        [string \":source (no file)\"]:9: in function 'F1'                                            \n        [string \":source (no file)\"]:12: in function 'handler'                                      \n        [string \":source (no file)\"]:15: in main chunk                                              \n```\n\n### Nvim version (nvim -v)\n\nhttps://github.com/neovim/neovim/commit/b34e137e43d359c8db4fb76028dea3b410842aff\n\n### Vim (not Nvim) behaves the same?\n\nNA\n\n### Operating system/version\n\nubuntu\n\n### Terminal name/version\n\ngnome-terminal\n\n### $TERM environment variable\n\nxterm-256color\n\n### Installation\n\nmake install",
    "closed_at": "2024-11-02T02:11:07Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/35768171?v=4",
        "events_url": "https://api.github.com/users/zeertzjq/events{/privacy}",
        "followers_url": "https://api.github.com/users/zeertzjq/followers",
        "following_url": "https://api.github.com/users/zeertzjq/following{/other_user}",
        "gists_url": "https://api.github.com/users/zeertzjq/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/zeertzjq",
        "id": 35768171,
        "login": "zeertzjq",
        "node_id": "MDQ6VXNlcjM1NzY4MTcx",
        "organizations_url": "https://api.github.com/users/zeertzjq/orgs",
        "received_events_url": "https://api.github.com/users/zeertzjq/received_events",
        "repos_url": "https://api.github.com/users/zeertzjq/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/zeertzjq/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/zeertzjq/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/zeertzjq",
        "user_view_type": "public"
    },
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "This may be a problem with other callbacks invoked using `pcall()` as well.",
            "created_at": "2024-11-01T03:15:12Z",
            "html_url": "https://github.com/neovim/neovim/issues/31019#issuecomment-2451214744",
            "id": 2451214744,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/31019",
            "node_id": "IC_kwDOAPphoM6SGpGY",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451214744/reactions"
            },
            "updated_at": "2024-11-01T03:15:12Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451214744",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/35768171?v=4",
                "events_url": "https://api.github.com/users/zeertzjq/events{/privacy}",
                "followers_url": "https://api.github.com/users/zeertzjq/followers",
                "following_url": "https://api.github.com/users/zeertzjq/following{/other_user}",
                "gists_url": "https://api.github.com/users/zeertzjq/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/zeertzjq",
                "id": 35768171,
                "login": "zeertzjq",
                "node_id": "MDQ6VXNlcjM1NzY4MTcx",
                "organizations_url": "https://api.github.com/users/zeertzjq/orgs",
                "received_events_url": "https://api.github.com/users/zeertzjq/received_events",
                "repos_url": "https://api.github.com/users/zeertzjq/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/zeertzjq/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/zeertzjq/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/zeertzjq",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "> This may be a problem with other callbacks invoked using `pcall()` as well.\n\nAha, looks that way. In the example, instead of a callback, doing\n```\nlocal ok,msg = pcall(handler)\nif not ok then SFP(\"ERROR: %s\", msg) end\n```\noutputs\n```\nERROR: [string \":source (no file)\"]:6: in F2()                                                      \n```",
            "created_at": "2024-11-01T03:24:45Z",
            "html_url": "https://github.com/neovim/neovim/issues/31019#issuecomment-2451222156",
            "id": 2451222156,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/31019",
            "node_id": "IC_kwDOAPphoM6SGq6M",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451222156/reactions"
            },
            "updated_at": "2024-11-01T03:24:45Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451222156",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/20450427?v=4",
                "events_url": "https://api.github.com/users/errael/events{/privacy}",
                "followers_url": "https://api.github.com/users/errael/followers",
                "following_url": "https://api.github.com/users/errael/following{/other_user}",
                "gists_url": "https://api.github.com/users/errael/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/errael",
                "id": 20450427,
                "login": "errael",
                "node_id": "MDQ6VXNlcjIwNDUwNDI3",
                "organizations_url": "https://api.github.com/users/errael/orgs",
                "received_events_url": "https://api.github.com/users/errael/received_events",
                "repos_url": "https://api.github.com/users/errael/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/errael/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/errael/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/errael",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "I don't know the ins and out, but doing\n```\nlocal ok,msg = xpcall(handler, debug.traceback)\nif not ok then print(string.format(\"ERROR: %s\", msg)) end\n```\nproduces\n```\n\nERROR: [string \":source (no file)\"]:6: in F2()                                                      \nstack traceback:                                                                                    \n^I[C]: in function 'error'                                                                          \n^I[string \":source (no file)\"]:6: in function 'F2'                                                  \n^I[string \":source (no file)\"]:9: in function 'F1'                                                  \n^I[string \":source (no file)\"]:12: in function <[string \":source (no file)\"]:11>                    \n^I[C]: in function 'xpcall'                                                                         \n^I[string \":source (no file)\"]:15: in main chunk                                                    \n```\nFound this in https://stackoverflow.com/questions/45788739/get-true-stack-trace-of-an-error-in-lua-pcall",
            "created_at": "2024-11-01T03:45:05Z",
            "html_url": "https://github.com/neovim/neovim/issues/31019#issuecomment-2451237462",
            "id": 2451237462,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/31019",
            "node_id": "IC_kwDOAPphoM6SGupW",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451237462/reactions"
            },
            "updated_at": "2024-11-01T03:45:05Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2451237462",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/20450427?v=4",
                "events_url": "https://api.github.com/users/errael/events{/privacy}",
                "followers_url": "https://api.github.com/users/errael/followers",
                "following_url": "https://api.github.com/users/errael/following{/other_user}",
                "gists_url": "https://api.github.com/users/errael/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/errael",
                "id": 20450427,
                "login": "errael",
                "node_id": "MDQ6VXNlcjIwNDUwNDI3",
                "organizations_url": "https://api.github.com/users/errael/orgs",
                "received_events_url": "https://api.github.com/users/errael/received_events",
                "repos_url": "https://api.github.com/users/errael/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/errael/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/errael/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/errael",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 3,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/31019/comments",
    "created_at": "2024-11-01T03:09:45Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/31019/events",
    "html_url": "https://github.com/neovim/neovim/issues/31019",
    "id": 2628213472,
    "labels": [
        {
            "color": "f9d0c4",
            "default": true,
            "description": "issues reporting wrong behavior",
            "id": 77997474,
            "name": "bug",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3NA==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": "stdlib",
            "id": 573222693,
            "name": "lua",
            "node_id": "MDU6TGFiZWw1NzMyMjI2OTM=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/lua"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/31019/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM6cp1rg",
    "number": 31019,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/31019/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "closed",
    "state_reason": "completed",
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/31019/timeline",
    "title": "vim.on_key callback error display missing stack traceback",
    "updated_at": "2024-11-02T02:11:07Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/31019",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/20450427?v=4",
        "events_url": "https://api.github.com/users/errael/events{/privacy}",
        "followers_url": "https://api.github.com/users/errael/followers",
        "following_url": "https://api.github.com/users/errael/following{/other_user}",
        "gists_url": "https://api.github.com/users/errael/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/errael",
        "id": 20450427,
        "login": "errael",
        "node_id": "MDQ6VXNlcjIwNDUwNDI3",
        "organizations_url": "https://api.github.com/users/errael/orgs",
        "received_events_url": "https://api.github.com/users/errael/received_events",
        "repos_url": "https://api.github.com/users/errael/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/errael/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/errael/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/errael",
        "user_view_type": "public"
    }
}